main.tf

```diff
@@ -1,15 +1,17 @@
 provider "azurerm" {
-  version = "1.22.1"
+  features {}
 }
 
 locals {
-  tags = "${merge(var.common_tags,
-    map("Team Contact", "${var.team_contact}")
-    )}"
+  tags = merge(var.common_tags,
+    map("Deployment Environment", var.env),
+    map("Team Name", var.team_name),
+    map("Team Contact", var.team_contact)
+    )
```

Comment:
Can be changed as follows. Replace the variable name test with a suitable name. Check for syntax errors if any.

locals {
  test = {
    "Deployment Environment" = var.env
    "Team Name"    = var.team_name
    "Team Contact" = var.team_contact
    "Destroy Me"   = var.destroy_me
  }

  tags = merge(var.common_tags, local.test)
}

Reply:
suggested changes have been done